### ISSUE TITLE
"Allow Tool" problems in 1.3.X

### ISSUE THREAD

**Issue by kertiek** [cseelhoff/RimThreaded#439]
**Describe the bug**
allow tool gui exception when selecting multiple objects/corpses on the ground 

![image](https://user-images.githubusercontent.com/21158056/106901422-811b0c00-66d6-11eb-98f9-7e12ef002efe.png)

**To Reproduce (VERY IMPORTANT)**
1 go to architect -> orders -> allow tool

2 try to unforbid all the insect corpses on the ground

3 read the gui exception

log
https://gist.github.com/3b8828fd4f6128f5dabbcd333fa1b45a

save ready to reproduce, just select the corpses on the ground:

[vanilla rt allow tool gui exception.zip](https://github.com/cseelhoff/RimThreaded/files/5926119/vanilla.rt.allow.tool.gui.exception.zip)

also in the same log there was a friendly fire exception while i was shooting at the insects

```
Exception ticking Bullet_ChargeRifle174728 (at (222, 0, 111)): System.NullReferenceException: Object reference not set to an instance of an object
  at RimThreaded.Projectile_Patch.CanHit (Verse.Projectile __instance, System.Boolean& __result, Verse.Thing thing) [0x00001] in <0f61a760d1dc4edc9c0c95583064ca6f>:0 
  at (wrapper dynamic-method) Verse.Projectile.Verse.Projectile.CanHit_Patch1(Verse.Projectile,Verse.Thing)
  at RimThreaded.Projectile_Patch.CheckForFreeIntercept (Verse.Projectile __instance, System.Boolean& __result, Verse.IntVec3 c) [0x0008d] in <0f61a760d1dc4edc9c0c95583064ca6f>:0 
  at (wrapper dynamic-method) Verse.Projectile.Verse.Projectile.CheckForFreeIntercept_Patch1(Verse.Projectile,Verse.IntVec3)
  at (wrapper delegate-invoke) <Module>.invoke_callvirt_bool_Projectile_IntVec3(Verse.Projectile,Verse.IntVec3)
  at RimThreaded.Projectile_Patch.CheckForFreeInterceptBetween (Verse.Projectile __instance, System.Boolean& __result, UnityEngine.Vector3 lastExactPos, UnityEngine.Vector3 newExactPos) [0x000ed] in <0f61a760d1dc4edc9c0c95583064ca6f>:0 
  at (wrapper dynamic-method) Verse.Projectile.Verse.Projectile.CheckForFreeInterceptBetween_Patch1(Verse.Projectile,UnityEngine.Vector3,UnityEngine.Vector3)
  at Verse.Projectile.Tick () [0x00065] in <d72310b4d8f64d25aee502792b58549f>:0 
  at RimThreaded.RimThreaded.ExecuteTicks () [0x00105] in <0f61a760d1dc4edc9c0c95583064ca6f>:0 
Verse.Log:Error(String, Boolean)
RimThreaded.RimThreaded:ExecuteTicks()
RimThreaded.RimThreaded:ProcessTicks(ThreadInfo)
RimThreaded.RimThreaded:InitializeThread(ThreadInfo)
RimThreaded.<>c__DisplayClass80_0:<CreateWorkerThread>b__0()
System.Threading.ThreadHelper:ThreadStart_Context(Object)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
System.Threading.ThreadHelper:ThreadStart()

```

**Comment by cseelhoff** [comment]
seems to be fixed in 1.3.0.9 - please reopen if issue still occurs

**Comment by Niilo007** [comment]
i got some strange stuff related to allow tool.

https://gist.github.com/HugsLibRecordKeeper/1aaa681995168cc9064dc3b52aad03ee
